schematic/schemas/json_schema_generator.py

```diff
@@ -0,0 +1,747 @@
+            self.current_node = None
+
+    def are_nodes_remaining(self) -> bool:
+        """
+        Returns:
+            Whether or not there are any nodes left to process
+        """
+        return self.current_node is not None
+
+    def is_current_node_processed(self) -> bool:
+        """
+        Returns:
+            Whether or not the current node has been processed yet
+        """
+        return self.current_node in self.processed_nodes
+
+    def update_range_domain_map(
+        self, node_display_name: str, node_range_display_names: list[str]
+    ) -> None:
+        """Updates the range domain map
+
+        Arguments:
+            node_display_name: The display name of the node
+            node_range_display_names: The display names of the the nodes range
+        """
+        for node in node_range_display_names:
+            if node not in self.range_domain_map:
+                self.range_domain_map[node] = []
+            self.range_domain_map[node].append(node_display_name)
+
+    def update_reverse_dependencies(
+        self, node_display_name: str, node_dependencies_display_names: list[str]
+    ) -> None:
+        """Updates the reverse dependencies
+
+        Arguments:
+            node_display_name: The display name of the node
+            node_dependencies_display_names: the display names of the reverse dependencies
+        """
+        for dep in node_dependencies_display_names:
+            if dep not in self.reverse_dependencies:
+                self.reverse_dependencies[dep] = []
+            self.reverse_dependencies[dep].append(node_display_name)
+
+    def update_nodes_to_process(self, nodes: list[str]) -> None:
+        """Updates the nodes to process with the input nodes
+
+        Arguments:
+            nodes: Nodes to add
+        """
+        self.nodes_to_process += nodes
+
+    def update_processed_nodes_with_current_node(self) -> None:
+        """Adds the current node to the list of processed nodes"""
+        if self.current_node is None:
+            raise ValueError("Current node is None")
+        self.processed_nodes.append(self.current_node)
+
+
+class JSONSchemaGenerator:  # pylint: disable=too-few-public-methods
+    "Data Model Json Schema"
+
+    def __init__(
+        self,
+        jsonld_path: str,
+        graph: nx.MultiDiGraph,
+    ) -> None:
+        self.jsonld_path = jsonld_path
+        self.graph = graph
+        self.dmge = DataModelGraphExplorer(self.graph)
+        self.dmr = DataModelRelationships()
+        self.rel_dict = self.dmr.relationships_dictionary
+
+    def create_json_schema(
+        self,
+        datatype: str,
+        schema_name: str,
+        schema_path: Union[str, None] = None,
+        write_schema: bool = True,
+    ) -> dict[str, Any]:
+        """
+        Consolidated method that aims to gather dependencies and value constraints across terms
+        / nodes in a schema.org schema and store them in a jsonschema /JSON Schema schema.
```

Comment:
Consider writing the docstrings without the previous state that the changes were made from in mind. I assume that "Consolidated method...." is meaning that this is taking a handful of functionality and squashing it together, but if I were to read this a year from now - I don't care that it was constructed from several functions. Rather, I care about what the function does today and right now.